packages/discount-code-importer/src/main.js

```diff
@@ -1,3 +1,221 @@
-export default function findSum (a, b) {
-  return a + b
+/* @flow */
+import type {
+  LoggerOptions,
+  ChunkOptions,
+  CodeDataArray,
+  CodeData,
+  ConstructorOptions,
+  Summary,
+} from 'types/discountCodes'
+import type {
+  Client,
+  AuthMiddlewareOptions,
+  SyncAction,
+ } from 'types/sdk'
+import npmlog from 'npmlog'
+import Promise from 'bluebird'
+import _ from 'lodash'
+import { createClient } from '@commercetools/sdk-client'
+import { createRequestBuilder } from '@commercetools/api-request-builder'
+import { createAuthMiddlewareForClientCredentialsFlow }
+  from '@commercetools/sdk-middleware-auth'
+import { createHttpMiddleware } from '@commercetools/sdk-middleware-http'
+import { createUserAgentMiddleware }
+  from '@commercetools/sdk-middleware-user-agent'
+import { createSyncDiscountCodes } from '@commercetools/sync-actions'
+import { version } from '../package.json'
+
+export default class DiscountCodeImport {
+  // Set flowtype annotations
+  batchSize: number;
+  continueOnProblems: boolean;
+  client: Client;
+  apiConfig: AuthMiddlewareOptions;
+  logger: LoggerOptions;
+  _summary: Summary;
+  syncDiscountCodes: SyncAction;
+
+  constructor (
+    logger: LoggerOptions,
+    options: ConstructorOptions,
+  ) {
+    this.batchSize = options.batchSize || 50
+    this.apiConfig = options.apiConfig
+    this.continueOnProblems = options.continueOnProblems || false
+    this.client = createClient({
+      middlewares: [
+        createAuthMiddlewareForClientCredentialsFlow(this.apiConfig),
+        createUserAgentMiddleware({
+          libraryName: 'discount-code-importer',
+          libraryVersion: version,
```

Comment:
I would prefer a context given on the `version` as variable, because it is easy to think that this is an env variable

```
import pkg from 'package.json'

pkg.version // <-- use this, give it context
```